packages/office-ui-fabric-react/src/components/CommandBar/examples/CommandBar.Customization.Example.tsx

```diff
@@ -0,0 +1,137 @@
+
+  public render() {
+    return (
+      <div>
+        <CommandBar
+          isSearchBoxVisible={ false }
+          items={
+            [
+              {
+                key: 'new',
+                name: 'Add',
+                onRender: this._renderSplitButtonMenuItem,
+                className: 'ms-CommandBarItem',
+                subMenuProps: {
+                  items: [
+                    {
+                      key: 'emailMessage',
+                      name: 'Email message',
+                      icon: 'Mail',
+                      ['data-automation-id']: 'newEmailButton'
+                    },
+                    {
+                      key: 'calendarEvent',
+                      name: 'Calendar event',
+                      icon: 'Calendar',
+                      ['data-automation-id']: 'newCalendarButton'
+                    }
+                  ],
+                },
+              },
+            ]
+          }
+        />
+      </div>
+    );
+  }
+
+  @autobind
+  private _renderSplitButtonMenuItem(item: IContextualMenuItem) {
+    let darkerBG = this.state.isContextMenuShown && styles.darkerBG;
+    let leftIconClassNames = css(
+      styles.icon,
+      styles.themeDarkAltColor,
+      'ms-Icon ms-Icon--Add',
+      darkerBG);
+    let dropDownIconClassNames = css(
+      styles.icon,
+      'ms-Icon ms-Icon--ChevronDown',
+      darkerBG);
+    let leftTextClassNames = css(
+      styles.leftText,
+      'ms-CommandBarItem-commandText',
+      darkerBG);
+
+    let containerClasName = css(
+      styles.customButtonContainer,
+      darkerBG
+    );
+
+    let dropDownButtonClass = css(
+      styles.button,
+      darkerBG
+    );
+    let mainBtnClassName = css(
+      !item.name && ('ms-CommandBarItem--noName'),
+      styles.button,
+      darkerBG
+    );
+
+    return (
+      <div>
+        <div className={ containerClasName } ref={ ref => this.container = ref }>
+          <Button
+            key="mainButton"
+            buttonType={ ButtonType.normal }
+            onClick={ () => { } }
+            className={ mainBtnClassName }
+            data-is-focusable={ true }>
+            <span className={ leftIconClassNames } />
+            <span className={ leftTextClassNames }>{ "New" }</span>
+          </Button>
+          <span className={ styles.splitter }>|</span>
+          <Button
+            key='dropDownButton'
+            buttonType={ ButtonType.normal }
+            onClick={ this.onClickChevron }
+            className={ dropDownButtonClass }>
+            <span
+              className={ dropDownIconClassNames }
+              data-is-focusable={ true } />
+          </Button>
+        </div>
+        {
+          this.state && this.state.isContextMenuShown &&
+          <ContextualMenu
+            isBeakVisible={ true }
+            className={ css('ms-CommandBar-menuHost') }
+            items={ item.subMenuProps.items }
+            targetElement={ this.container }
```

Comment:
Hmm, I thought targetElement was deprecated and that you can just use target (and put an id on the button to identify it.)